src/modules/airspeed_selector/airspeed_selector_main.cpp

```diff
@@ -593,9 +593,14 @@ void AirspeedModule::select_airspeed_and_publish()
 
 	// print warning or info, depending of whether airspeed got declared invalid or healthy
 	if (_valid_airspeed_index != _prev_airspeed_index &&
-	    (_number_of_airspeed_sensors > 0 || !_vehicle_land_detected.landed) &&
-	    _valid_airspeed_index != _prev_airspeed_index) {
-		if (_prev_airspeed_index > 0) {
+	    _number_of_airspeed_sensors > 0) {
+		if (_vehicle_status.arming_state == !vehicle_status_s::ARMING_STATE_ARMED && _prev_airspeed_index > 0) {
+			mavlink_log_critical(&_mavlink_log_pub,
+					     "Airspeed sensor failure detected. Check connection between sensor and flight controller.\t");
```

Comment:
```suggestion
					     "Airspeed sensor failure detected. Check connection and reboot\t");
```
By default the airspeed validator doesn't declare the sensor valid again if the data was once dropped, so if the user will want to arm a reboot first is required.

Reply:
addressed with the last commit, thx for having a look!